plugins/woocommerce-blocks/.wp-env.json

```diff
@@ -1,5 +1,5 @@
 {
-	"core": "https://wordpress.org/wordpress-6.6-RC2.zip",
+	"core": "https://wordpress.org/wordpress-6.6.zip",
```

Comment:
Should we remove this line? In this way, our E2E tests will always pick the latest version of WordPress (including patch releases)

Reply:
The problem I see with that is that some E2E tests might start failing, and it might not be immediately obvious why because the WP version gets updated in the background. The current approach gives us a clear image of what is failing because of the release vs. what is failing because of us. 😄 Does that make sense?

Reply:
I agree with you, but we should also have a process in place to ensure this file is updated whenever a new patch release is out.

I prefer having a CI failure over risking unnoticed regressions due to a missed update of this file.

Reply:
Good point. Are you sure the patch releases are included, though? I haven't checked myself, but the readme states that [the latest **production** release will be used](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-env/#:~:text=The%20WordPress%20installation%20to,production%20release%20of%20WordPress.).

Reply:
I agree. Since a patch release is also a production release, we should be fine.

Reply:
Ah, sorry. For some reason, I read `patch` and thought `RC` releases. 😅 So, to finish my thought, we won't be testing against RCs anymore, which lets us catch production release bugs early.

Reply:
> we won't be testing against RCs anymore, which now lets us catch production release bugs early.
Yeah, you're right, but as I said earlier, without any process, there is the risk that this file will not be updated in time.

Also, WooCommerce Core and Gutenberg don't hardcode the version too:

- Gutenberg: https://github.com/WordPress/gutenberg/blob/47bde1a7696ceae94557dde3fa57cde3cfc9d84e/.wp-env.json#L4
- WooCommerce Core: https://github.com/woocommerce/woocommerce/blob/0c9e33bb7cb2b9091928339a02e5cf546a8a1b0e/plugins/woocommerce/.wp-env.json

Reply:
Can I just check what the context of this is:

> we won't be testing against RCs anymore, which now lets us catch production release bugs early.

How would we catch a production release bug early if we aren't testing against RCs?

Reply:
> Also, WooCommerce Core and Gutenberg don't hardcode the version too:

That's different than removing the entry, though. In the `.wp-env.json` config:
- `"core": null` (or empty) means automatic updates to production versions.
- `"core": "WordPress/WordPress"` means bleeding edge (synced with WP `trunk`).

Having said that, I wasn't aware that WC Core is at the bleeding edge as well. It's a pretty strong argument for the Blocks to do the same if you ask me! 😄

Reply:
> How would we catch a production release bug early if we aren't testing against RCs?

Right now (in WC Blocks), we do because we manually update to the RCs. If we only update to production releases, that might become an issue. I'm wondering about the disconnect between the WC Core and Blocks test environments, though: Shouldn't we be running our E2E tests against the same environment, at least in terms of the WP version? Currently, IIUC, Core is set to run against [the production release of WP](https://github.com/woocommerce/woocommerce/blob/0c9e33bb7cb2b9091928339a02e5cf546a8a1b0e/plugins/woocommerce/.wp-env.json), and Blocks use the RC if available.

cc: @adimoldovan

Reply:
We have a to-do to add the RC as a daily check for Core e2e.
We're currently using the latest WP in the PR checks and also check latest-1 with a daily run.

Reply:
> We have a to-do to add the RC as a daily check for Core e2e.

So, if we merge this PR we run each PR against the latest WP, but we plan to run against the RC and latest-1 as a daily check? If this is the plan then that sounds pretty good!

Reply:
That's the plan for Core e2e. Block e2e are only running for PRs currently, but I think we should include them in a daily check too, with Latest-1.
But back to this PR, I think it's safe to merge, we should use the latest for PRs.

Reply:
Let's merge this PR since we should already be on v6.6.1 and continue the discussion in a separate issue. How does that sound?

Reply:
Here's the issue https://github.com/woocommerce/woocommerce/issues/50163

Reply:
Sorry for going back and forth here, but it occurred to me that it indeed makes more sense to keep the WP auto-updated and only hardcode RCs when released. As @gigitux said above, it's better to have tests failing and not know why than to miss an actual regression! I've already updated the issue.